packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts

```diff
@@ -73,4 +81,36 @@ test.describe('VideoGallery tests', async () => {
       'video-gallery-with-1-joining-1-hold-gridview-participant.png'
     );
   });
+
+  /* @conditional-compile-remove(pinned-participants) */
+  test('Remote video tile pin menu button should be disabled when max remote video tiles are pinned', async ({
+    page,
+    serverUrl
+  }, testInfo) => {
+    const displayNames = ['Tony Hawk', 'Marie Curie', 'Gal Gadot', 'Margaret Atwood', 'Kobe Bryant', "Conan O'Brien"];
+    const participants = displayNames.map((name) => defaultMockRemoteParticipant(name));
+    const initialState = defaultMockCallAdapterState(participants);
+
+    await page.goto(buildUrlWithMockAdapter(serverUrl, initialState));
+
+    if (isTestProfileMobile(testInfo)) {
+      // TODO: Test that pin menu item is disabled when maximum remote VideoTiles are pinned in VideoGallery when
+      // drawer menu on long touch has been implemented
+      return;
+    }
+
+    // pin remote video tiles up to the max allowed in the call composite
+    for (let i = 0; i < 4; i++) {
+      // click the last more button on the page which is presumably on an unpinned remote video tile
+      const moreOptionsButton = page.locator(dataUiId('video-tile-more-options-button')).last();
+      await screenshotOnFailure(page, async () => await moreOptionsButton.click({ timeout: perStepLocalTimeout() }));
```

Comment:
we can use `waitforselector` helper method to avoid this repetitive code as it takes care of screenshot on failure